smach_viewer/package.xml

```diff
@@ -25,14 +25,22 @@
   <build_depend>rostest</build_depend>
 
   <!-- Dependencies needed after this package is compiled. -->
-  <run_depend>smach_ros</run_depend>
-  <run_depend>smach_msgs</run_depend>
+  <exec_depend>smach_ros</exec_depend>
+  <exec_depend>smach_msgs</exec_depend>
   <!-- originally depends on xdot : https://github.com/jbohren/xdot/blob/2.0.1/package.xml#L25-L26 -->
-  <run_depend>wxpython</run_depend>
-  <run_depend>graphviz</run_depend>
-  <run_depend>python-gi</run_depend>
-  <run_depend>python-gi-cairo</run_depend>
-  <run_depend>gtk3</run_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">wxpython</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-xdot</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-wxgtk4.0</exec_depend>
+  <exec_depend>graphviz</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-gi</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-gi</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-rospkg</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-rospkg</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-qt5-bindings</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-qt5-bindings</exec_depend>
+  <exec_depend>python-gi-cairo</exec_depend>
+  <!-- <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-gi-cairo</exec_depend> -->
```

Comment:
This is actually the right syntax. Why is it commented?

Reply:
Because the actual key did not exist, but I see you created a pull request for that.

Reply:
Because the python3-gi-cairo key did not exist in ros-repo, but I see that you added them in a PR. Thanks for that!

Reply:
No problem, xdot still pending: https://github.com/ros/rosdistro/pull/32004